src/commands/sites/sites-create.js

```diff
@@ -85,6 +85,19 @@ const sitesCreate = async (options, command) => {
   }
 
   const { name: nameFlag } = options
+
+  // netlify sites:create --name
+  if (nameFlag === true) {
+    error('Please specify site name, example: --name <site_name>')
+    return
+  }
```

Comment:
I think instead of this custom code it would be better to change the `commander` configuration at the end of this file to make the argument required:
```diff
-    .option('-n, --name [name]', 'name of site')
-    .option('-a, --account-slug [slug]', 'account slug to create the site under')
+    .option('-n, --name <name>', 'name of site')
+    .option('-a, --account-slug <slug>', 'account slug to create the site under')
```

This won't allow `--name` and  `--account-slug` to be used without argument.